Tests/AsyncClientTests.swift

```diff
@@ -0,0 +1,310 @@
+// Copyright © 2023 DIAMIR. All Rights Reserved.
+
+@testable import Endpoints
+import XCTest
+
+class AsyncClientTests: XCTestCase {
+    var tester: AsyncClientTester<AnyClient>!
+
+    // 2021-02-12: redirect endpoints for httpbin.org no longer work, https://github.com/postmanlabs/httpbin/issues/617
+    let baseURL = URL(string: "https://nghttp2.org/httpbin/")!
+    override func setUp() {
+        tester = AsyncClientTester(test: self, client: AnyClient(baseURL: baseURL))
+    }
+
+    func testStatusErrorAsync() async throws {
+        do {
+            let call = AnyCall<DataResponseParser>(Request(.get, "status/400"))
+            _ = try await tester.test(call: call)
+            XCTFail("Should throw exception")
+        }
+        catch let endpointsError as EndpointsError {
+            XCTAssertEqual(endpointsError.error.localizedDescription, "bad request")
+        }
+    }
+
+    func testGetDataAsync() async throws {
+        let call = AnyCall<DataResponseParser>(Request(.get, "get"))
+        let (_, response) = try await tester.test(call: call)
+        XCTAssertEqual(response.statusCode, 200)
+    }
+
+    func testGetDataAsyncWithCancellation() async throws {
+        let exp = expectation(description: "task was executed until end")
+        exp.isInverted = true
+
+        let task = Task {
+            let call = AnyCall<DataResponseParser>(Request(.get, "get"))
+            _ = try await tester.test(call: call)
+            exp.fulfill()
+        }
+
+        try await Task.sleep(nanoseconds: 400)
+        task.cancel()
+        await fulfillment(of: [exp], timeout: 5)
+
+        XCTAssertTrue(task.isCancelled, "Parent Task should be cancelled.")
```

Comment:
not sure what this test case is testing. `task.isCancelled` will always be true because task.cancel() just sets this to true